closed/src/java.base/share/classes/com/sun/crypto/provider/NativeChaCha20Cipher.java

```diff
@@ -636,7 +636,10 @@ private void checkKeyAndNonce(byte[] newKeyBytes, byte[] newNonce)
     protected byte[] engineUpdate(byte[] in, int inOfs, int inLen) {
         byte[] out = new byte[inLen];
         try {
-            engine.doUpdate(in, inOfs, inLen, out, 0);
+            int size = engine.doUpdate(in, inOfs, inLen, out, 0);
+            // Special case for EngineAEADDec, doUpdate only buffers the input
+            // So the output array must be empty since no encryption happened yet
+            if(size == 0) return new byte[0];
```

Comment:
Pls add a space after `if`.
We'll need PRs for jdk14 and jdk as well.

Reply:
Done

Reply:
Don't put `if` and body on a single line, instead write:
```
if (size == 0) {
    return new byte[0];
}
```

Reply:
Done